src/updaters/dart/pubspec-yaml.ts

```diff
@@ -24,23 +24,30 @@ export class PubspecYaml extends DefaultUpdater {
    * @param {string} content The initial content
    * @returns {string} The updated content
    */
+
   updateContent(content: string): string {
-    const oldVersion = content.match(/version: ([0-9.]+)\+?([0-9]*$)/);
+    const oldVersion = content.match(/^version: ([0-9.]+)\+?(.*$)/m);
     let buildNumber = '';
 
     if (oldVersion) {
-      buildNumber = `${oldVersion[2]}`;
-      if (buildNumber.length > 0) {
+      buildNumber = oldVersion[2];
+      const parsedBuild = parseInt(buildNumber);
+      if (!isNaN(parsedBuild)) {
+        buildNumber = `+${parsedBuild + 1}`;
```

Comment:
I think this is a behavior change. I think it left the build number intentionally, but I don't know enough about the dart ecosystem to know if this is actually desired or not (the original author [intended it so](https://github.com/googleapis/release-please/pull/1053#discussion_r709567486)). cc @HelloCore

Reply:
I'd welcome feedback from @HelloCore here on intent, however, build number is not forced on the user if it does not exist (preserving existing behavior) - it only increments it if it does exist and is a number. When uploading apps to a store (eg., Google Play), build number can not be duplicated (each upload must have a different one) - there is a very good case for incrementing the build number as a result. It would introduce manual headache to *not* have build number updated.

Reply:
Actually, I'm okay with either keeping the old build number or increasing it because I mostly override the build number by using the CI build number (`flutter build ipa --build-number=1234`). 
I intended to keep the old build number here just because I don't want to make unnecessary changes on `pubspec.yaml`.

Reply:
Great, thanks for the input. We'll go ahead with this incremennt